eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java

```diff
@@ -115,14 +148,176 @@ private VariantType getVariantType(String reference, String alternate) {
         return variantType;
     }
 
+    private void merge(Long providedAccession, String hash, Long accessionInDatabase) {
+        Priority prioritised = prioritise(providedAccession, accessionInDatabase);
+
+        Query queryClustered = query(where(ACCESSION_KEY).is(prioritised.accessionToBeMerged));
+        List<? extends ClusteredVariantEntity> cvToMerge =
+                mongoTemplate.find(queryClustered, getClusteredVariantCollection(prioritised.accessionToBeMerged));
+
+        List<? extends ClusteredVariantEntity> cvToKeep =
+                mongoTemplate.find(query(where(ACCESSION_KEY).is(prioritised.accessionToKeep)),
+                                   getClusteredVariantCollection(prioritised.accessionToKeep));
+
+        if (isMultimap(cvToMerge) || isMultimap(cvToKeep)) {
```

Comment:
Should we also ensure that this policy of not merging or "merging into" overweight SNP is conformed to in the **historic data** in dbsnpCVE/dbsnpCVOE collections? 

Consider the following scenario:
**Historic data (before the current policy)**
dbSNP_RS1 was merged into dbSNP_RS0 - dbSNP_RS0 is an overweight SNP - both RS are in ASM1
Therefore, dbSNP_RS1 is in dbSNPCVOE and dbSNP_RS0 is in dbSNPCVE

**New data**
SS3 submitted to ASM1 - when clustering SS3, dbsnp_RS0  appears to be a merge target based on SS3's locus but we don't pursue the RS assignment because it is overweight. So we proceed to issue a new RS ID for SS3. 
Here, we are paying a price for a historic mistake since the correct course of action would have been to assign dbSNP_RS1 as the RS for SS3.

Reply:
thinking a bit more about this example, that's not exactly what would happen. Bear in mind that there are 2 things that can happen in the clustering pipeline: clustering (add a rs to a ss if it is unclustered) and RS merging (when an ss is already clustered in rs1, but the locus after remapping makes collision with rs2).

In your scenario, it would just cluster SS3 under RS0, but no further merging will happen. Although technically this doesn't go against the policy of "not merging into overweight", arguably it is kind of wrong too. However, if we ever split those overweight RS, no unmerging is needed,  (other than the dbsnp one that already happened during the import), just change the RS when the locus is different. I agree on discussing this in the next meeting.

In a slightly different scenario, if SS3 had RS3 (instead of being unclustered), nothing would be changed. SS3 would keep RS3, even though RS3 is not in ASM1 (also, it can't be in ASM1 because it would make collision with RS0).

Reply:
You are right, the terminology in my initial description does concern clustering. Nevertheless, IMO, RS0 should not be considered either as a merge target or as a RS assignment candidate.